### PR TITLE
docs - Added a note in how-tos/reset on how to use structuredClone

### PR DESCRIPTION
**Pull request by overthemike**
## Related Bug Reports or Discussions
N/A
Fixes #
N/A
## Summary
Added example on how to use structuredClone in v2 and the reasoning behind why it's different than v1.

## Check List

- [ x ] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

File: docs/how-tos/how-to-reset-state.mdx [hunk 1]
```diff
@@ -34,3 +34,34 @@ const reset = () => {
   state.obj = _.cloneDeep(initialObj)
 }
 ```
+
+> [!NOTE]
+> Using `structuredClone()`
+
+In 2022, there was a new global function added called `structuredClone` that is widely available in most modern browsers. While this function and `_.cloneDeep` from lodash seem to do the same thing, their implementations are quite different.
+
+In valtio v1, `proxy` cloned the object used for inital state under the hood. Because the original object isn't used, `structuredClone` works just fine as a drop in replacement for `_.cloneDeep`.
```
Comment (dai-shi) [inline docs/how-tos/how-to-reset-state.mdx:43]: Is this only the issue with `structuredClone`? Can it be the same with `_.cloneDeep`?

Comment (overthemike) [reply docs/how-tos/how-to-reset-state.mdx:43]: It doesn't seem to. I haven't tested very thoroughly but from what I've tested so far, it seems to work ok, even modifying the structure of the object doesn't really seem to affect it too much. This example doesn't seem to have any issues

```js
import { proxy, useSnapshot } from 'valtio'
import { cloneDeep } from 'lodash'

const initialState = {
  foo: {
    bar: {
      bat: 'bat initial'
    }
  }
}
const state = proxy({ obj: initialState })
const reset = () => {
  state.obj = cloneDeep(initialState)
}

function App() {
  const snap = useSnapshot(state)

  return (
    <div>
      <p>snap.foo.bar.bat: {snap.obj.foo.bar.bat}</p>
      <button onClick={() => state.obj = { foo: { bar: { asdf: 'bat2', ball: 'ball' } }}}>click</button>
      <button onClick={() => state.obj = { foo: { bar: 'bar2'}}}>click</button>
      <button onClick={() => reset()}>reset</button>
      <pre>{JSON.stringify(snap.obj, null, 2)}</pre>
    </div>
  )
}

export default App

```

Comment (overthemike) [reply docs/how-tos/how-to-reset-state.mdx:43]: Outputs as expected:

Initial state
```
snap.foo.bar.bat: bat initial

{
  "foo": {
    "bar": {
      "bat": "bat initial"
    }
  }
}
```

First button click
```
snap.foo.bar.bat:

{
  "foo": {
    "bar": {
      "asdf": "bat2",
      "ball": "ball"
    }
  }
}
```

Second button click
```
snap.foo.bar.bat:

{
  "foo": {
    "bar": "bar2"
  }
}
```

Reset button click
```
snap.foo.bar.bat: bat initial

{
  "foo": {
    "bar": {
      "bat": "bat initial"
    }
  }
}
```

Comment (dai-shi) [reply docs/how-tos/how-to-reset-state.mdx:43]: Can you try this?

```jsx
<button onClick={() => { state.obj.foo.bar.bat = 'bat changed' }}>click</button>
```

Comment (overthemike) [reply docs/how-tos/how-to-reset-state.mdx:43]: yeah that does mess it up. I'll update the PR once I get a chance.

Comment (overthemike) [reply docs/how-tos/how-to-reset-state.mdx:43]: I updated the docs. Let me know if you want anything reworded.

File: docs/how-tos/how-to-reset-state.mdx [hunk 2]
```diff
@@ -34,3 +34,36 @@ const reset = () => {
   state.obj = _.cloneDeep(initialObj)
```
Comment (dai-shi) [inline docs/how-tos/how-to-reset-state.mdx:34]: I mean this is wrong too. L31-35

Can you please fix the doc in this way?
- Fix cloneDeep usage with v2.
- Add a note about structuredClone

Speaking of v2, we now have `deepClone` from valtio/utils, we don't need to use cloneDeep.
So, we could restructure the docs to use `deepClone` first and `structuredClone` second.
(I thought about explaining structuredClone first, but `deepClone` is capable of avoid copying `ref()`'d object.

Comment (overthemike) [reply docs/how-tos/how-to-reset-state.mdx:34]: Sure thing. I'll push up the changes soon.

Comment (overthemike) [reply docs/how-tos/how-to-reset-state.mdx]: Changes have been pushed. Let me know of any feedback.

**Review by dai-shi — APPROVED** [review]
Looks perfect!

**Pull request merged**